packages/@ama-sdk/create/src/index.ts

```diff
@@ -69,26 +66,31 @@ const schematicArgs = [
   '--name', name,
   '--package', pck,
   '--package-manager', packageManager,
-  '--directory', targetDirectory,
-  ...(argv['spec-path'] ? ['--spec-path', argv['spec-path']] : []),
   ...(typeof argv['o3r-metrics'] !== 'undefined' ? [`--${!argv['o3r-metrics'] ? 'no-' : ''}o3r-metrics`] : [])
 ];
 
-const getSchematicStepInfo = (schematic: string) => ({
-  args: [binPath, schematic, ...schematicArgs]
-});
+const resolveTargetDirectory = resolve(process.cwd(), targetDirectory);
 
 const run = () => {
+  const isSpecPathUrl = url.URL.canParse(argv['spec-path']);
 
   const runner = process.platform === 'win32' ? `${packageManager}.cmd` : packageManager;
   const steps: { args: string[]; cwd?: string; runner?: string }[] = [
-    getSchematicStepInfo(schematicsToRun[0]),
+    { args: [binPath, `${schematicsPackage}:typescript-shell`, ...schematicArgs, '--directory', targetDirectory] },
     ...(
       packageManager === 'yarn'
-        ? [{ runner, args: ['set', 'version', getYarnVersion()], cwd: resolve(process.cwd(), targetDirectory)}]
+        ? [{ runner, args: ['set', 'version', getYarnVersion()], cwd: resolveTargetDirectory }]
         : []
     ),
-    ...schematicsToRun.slice(1).map(getSchematicStepInfo)
+    ...(argv['spec-path'] ? [{
+      args: [
+        binPath,
+        `${schematicsPackage}:typescript-core`,
+        ...schematicArgs,
+        '--spec-path', isSpecPathUrl ? argv['spec-path'] : relative(join(process.cwd(), targetDirectory), join(process.cwd(), argv['spec-path']))
```

Comment:
```suggestion
        '--spec-path', isSpecPathUrl ? argv['spec-path'] : relative(resolveTargetDirectory, join(process.cwd(), argv['spec-path']))
```
(my bad, I could have seen it in my first suggestion)